generators/example/src/main/scala/ConfigMixins.scala

```diff
@@ -29,110 +30,78 @@ import ConfigValName._
 // Common Parameter Mixins
 // -----------------------
 
-/**
- * Class to specify where the BootRom file is (from `rebar` top)
- */
 class WithBootROM extends Config((site, here, up) => {
   case BootROMParams => BootROMParams(
     contentFileName = s"./bootrom/bootrom.rv${site(XLen)}.img")
 })
 
-// DOC include start: WithGPIO
-/**
- * Class to add in GPIO
- */
 class WithGPIO extends Config((site, here, up) => {
   case PeripheryGPIOKey => List(
     GPIOParams(address = 0x10012000, width = 4, includeIOF = false))
+  case BuildTop => (clock: Clock, reset: Bool, p: Parameters, success: Bool) => {
+    val top = up(BuildTop, site)(clock, reset, p, success)
```

Comment:
This is cute.

Reply:
Will this cause problems is someone includes multiple of the same mixin?
It seems to work as long as each buildtop method is idempotent, but I'm not sure thats the case. I.e. if any buildtop methods construct hardware rather than just making connections you might have problems.

Reply:
Yeah you are right non-idempotent mixins will cause strange behavior with this. Is that a issue we should be concerned about?

Reply:
As long as non of our mixins are currently non-idempotent then I think just the warning is fine. Thanks.